flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/DataMapManager.java

```diff
@@ -538,7 +538,7 @@ private void genSourceFromMappingThroughElements(MappingElements mappingElements
             String lastSplit = splits[length - 1];
             if (length == 1 && lastSplit.matches("^-?\\d+$")) {
```

Comment:
btw, why are we using a regex expression to check if this string is a number?

Reply:
Cases like `students.1`, we have to identify it as array of `students`. That's why we split port and check the last element.